arwen/elrondapi/bigFloatOps.go

```diff
@@ -481,21 +521,29 @@ func v1_4_bigFloatPow(context unsafe.Pointer, destinationHandle, opHandle, expon
 	lengthOfResult := big.NewInt(0).Div(big.NewInt(0).Mul(op2BigInt, big.NewInt(int64(opBigInt.BitLen()))), big.NewInt(8))
 	managedType.ConsumeGasForThisBigIntNumberOfBytes(lengthOfResult)
 
-	powResult := new(big.Float).Set(pow(op, exponent))
+	powResult, err := pow(context, op, exponent)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return
+	}
 	setResultIfNotInfinity(arwen.GetVMHost(context), powResult, destinationHandle)
 }
 
-func pow(base *big.Float, exp int32) *big.Float {
+func pow(context unsafe.Pointer, base *big.Float, exp int32) (*big.Float, error) {
 	result := big.NewFloat(1)
 	result.SetPrec(base.Prec())
+	runtime := arwen.GetRuntimeContext(context)
 	for i := 0; i < int(exp); i++ {
-		result.Mul(result, base)
+		resultMul, err := arwenMath.MulBigFloat(result, base)
+		if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
```

Comment:
Is this `WithFault` necessary here? Could it be just `if err != nil`?

Reply:
yes, the error returned from `arwenMath.MulBigFloat(result, base)` is basically the panic handled by the defer-recover mechanism.